drivers/display/ssd1306.c

```diff
@@ -373,23 +376,47 @@ static int ssd1306_set_orientation(const struct device *dev,
 static int ssd1306_set_pixel_format(const struct device *dev,
 				    const enum display_pixel_format pf)
 {
-	if (pf == PIXEL_FORMAT_MONO10) {
+	struct ssd1306_data *data = dev->data;
+
+	uint8_t cmd;
+	int ret;
+
+	if (pf == data->pf) {
 		return 0;
 	}
-	LOG_ERR("Unsupported");
-	return -ENOTSUP;
+
+	if (pf == PIXEL_FORMAT_MONO10) {
+		cmd = SSD1306_SET_REVERSE_DISPLAY;
+	} else if (pf == PIXEL_FORMAT_MONO01) {
+		cmd = SSD1306_SET_NORMAL_DISPLAY;
+	} else {
+		LOG_ERR("Unsupported");
```

Comment:
This is actually not an error, should be LOG_INF or LOG_DBG.

Reply:
> This is actually not an error, should be LOG_INF or LOG_DBG.

`LOG_DBG("Unsupported pixel format");`

Reply:
I don't quite agree on this not being an error, but I've changed it to your suggestion.

Reply:
Please explain your concern.

There is no functional error, because it keeps the default setting, not supported pixel format does not change the display functionality. From a driver API perspective, not supporting a particular pixel format is not an error.

Reply:
My reasoning is, that if I return an error (`-ENOTSUP`), I should also log it as an error. Also in the rest of the file, `LOG_ERR` is used for this purpose most of the time.

However, I have now read the description in log.h. There it says, that `LOG_ERR` is used for unrecoverable errors, which supports your complaint. Maybe `LOG_WRN` would be an even better choice than `_DBG`, though.

Reply:
> However, I have now read the description in log.h. There it says, that `LOG_ERR` is used for unrecoverable errors, which supports your complaint. Maybe `LOG_WRN` would be an even better choice than `_DBG`, though.

okay, I am fine with LOG_WRN().

Reply:
great, done